cmd/kubebuilder/initproject/vendor.go

```diff
@@ -92,16 +85,94 @@ func depExists() bool {
 	return err == nil
 }
 
-var depManifestTmpl = fmt.Sprintf("%s\n%s", depManifestHeader, depManifestOverride)
+func createNewDepManifest() {
+	depTmplArgs := map[string]string{
+		"Version": version.GetVersion().KubeBuilderVersion,
+	}
+	depManifestTmpl := fmt.Sprintf("%s\n%s\n%s", depManifestHeader, depManifestKBMarker, depManifestOverride)
+	util.Write(depManifestFile, "dep-manifest-file", depManifestTmpl, depTmplArgs)
+}
+
+// updateDepManifest updates the existing dep manifest with newer dependencies.
+// dep manifest update workflow:
+// Try to read user managed dep manifest section. If success, then append the
+// user managed dep with KB managed section and update the dep Manifest.
+func updateDepManifest() error {
+	// open the existing dep manifest.
+	f, err := os.Open(depManifestFile)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
```

Comment:
I suggest to move this block to function `tryReadingUserDeps`. You only need to open and close this file in that function.
 ```
+	f, err := os.Open(depManifestFile)
+	if err != nil {
+		return err
+	}
+	defer f.Close()```

Reply:
The only reason I kept it this way is to make `tryReadingUserDeps` unit-testable. I think if we had fakeFS, it would have made testing a bit easier for the whole fn.